### PR TITLE
Add SOA sample doc

### PR DESCRIPTION
**Pull request by rosemarymarano**
First pass at new SOA sample doc.
Please add Rupesh Das as a reviewer.

### CODE REVIEW

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+| `t3PublicAddress` | Public address for the T3 channel.  This should be set to the public address of the Kubernetes cluster.  This would typically be a load balancer address. <p/>For development environments only: In a single server (all-in-one) Kubernetes deployment, this may be set to the address of the master, or at the very least, it must be set to the address of one of the worker nodes. | If not provided, the script will attempt to set it to the IP address of the Kubernetes cluster |
+| `weblogicCredentialsSecretName` | Name of the Kubernetes secret for the Administration Server's user name and password. If not specified, then the value is derived from the `domainUID` as `<domainUID>-weblogic-credentials`. | `soainfra-domain-credentials` |
+| `weblogicImagePullSecretName` | Name of the Kubernetes secret for the Docker Store, used to pull the WebLogic Server image. |   |
+| `rcuSchemaPrefix` | The schema prefix to use in the database, for example `SOA1`.  You may wish to make this the same as the domainUID in order to simplify matching domains to their RCU schemas. | `SOA1` |
+| `rcuDatabaseURL` | The database URL. | `soadb.soans:1521/soapdb.us.oracle.com` |
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Please change this line
| `rcuDatabaseURL` | The database URL. | `soadb.soans:1521/soapdb.us.oracle.com` |
to
| `rcuDatabaseURL` | The database URL. | `soadb.soans:1521/soapdb.my.domain.com` |

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+| `rcuCredentialsSecret` | The Kubernetes secret containing the database credentials. | `soainfra-rcu-credentials` |
+
+In addition, you should make a note of the environment variable `DOMAIN_TYPE` in the `create-domain-job-template.yaml` file, in the same directory as the `create-domain-inputs.yaml` file.
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: The content in lines 220 and 221 is obsolete now as we are programmatically updating the value of DOMAIN_TYPE variable. Hence please remove these two lines.

**Comment by rosemarymarano** [comment]
Hi Samba,

I have incorporated the edits you requested in the latest commit.

Rosemary

**Comment by rjeberhard** [comment]
Hi, @rosemarymarano.  I see that you've incorporated the edits.  Is this ready to be further reviewed and merged?

**Comment by markxnelson** [comment]
No @rjeberhard, further review is needed, some of the reviewers are tied up with OOW tasks.  I would ask that this be left to sit until we get all the key reviewers' feedback please.

**Comment by rjeberhard** [comment]
Fine by me, @markxnelson...  I just wanted to make sure that if it was ready then it wasn't held up waiting to be merged.

**Comment by rosemarymarano** [comment]
Thanks for checking @rjeberhard; @marknelson is right; there is still more work to do on it. I will remove WIP from the title when it is ready to merge.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 3]
```diff
@@ -0,0 +1,605 @@
...
+description: "Sample for creating a SOA Suite domain home on an existing PV or
+PVC, and the domain resource YAML file for deploying the generated SOA domain."
+---
+
+
+The sample scripts demonstrate the creation of a SOA Suite domain home on an
+existing Kubernetes persistent volume (PV) and persistent volume claim (PVC). The scripts
+also generate the domain YAML file, which can then be used to start the Kubernetes
+artifacts of the corresponding domain. Optionally, the scripts start up the domain,
+and SOA server pods and services based on the domain type (for example, soa, osb, soaosb, soaess and soaessosb).
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Please remove below statement:
"Optionally, the scripts start up the domain,
and SOA server pods and services based on the domain type (for example, soa, osb, soaosb, soaess and soaessosb)."

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+
+#### Prerequisites
+
+Before you begin, read this document, [Domain resource]({{< relref "/userguide/managing-domains/domain-resource/_index.md" >}}).
+
+The following prerequisites must be handled prior to running the create domain script:
+
+* Make sure that Kubernetes is set up in the environment. For details, see [Set Up Kubernetes]({{< relref "/userguide/overview/k8s-setup.md" >}}).
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Please update "Set Up Kubernetes" to "Kubernetes setup guide"

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+* Make sure that the WebLogic operator is running. See [Manage operators]({{< relref "/userguide/managing-operators/_index.md" >}}) for operator infrastructure setup and [Install the operator]({{< relref "/userguide/managing-operators/installation/_index.md" >}}) for operator installation.
+* The operator requires SOA Suite 12.2.1.3.0 with patch 29135930 applied. For details on how to obtain or create the image, see [SOA domains]({{< relref "/userguide/managing-domains/soa-suite/_index.md#obtaining-the-soa-suite-docker-image" >}}).
+* Create a Kubernetes namespace (for example, `soans`) for the domain unless you intend to use the default namespace. Use the same namespace in all the other steps.
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Please update
Use the same namespace in all the other steps.
to 
Use the newly created namespace in all the following steps.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 6]
```diff
@@ -0,0 +1,605 @@
...
+For details, see [Prepare to run a domain]({{< relref "/userguide/managing-domains/prepare.md" >}}).
+
+  ```
+$ kubectl create namespace soans
+  ```
+
+* In the same Kubernetes namespace, follow these instructions to create the Kubernetes persistent volume (PV) where the domain home will be hosted,
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Please remove lines 32 to 97. Then add below latest content for this step:

In the same Kubernetes namespace created above, the user needs to create the Kubernetes persistent volume (PV) where the domain home will be hosted, and the Kubernetes persistent volume claim (PVC) for the domain.

To create PV/PVC run the create-pv-pvc.sh script located at kubernetes/samples/scripts/create-weblogic-domain-pv-pvc. Please follow the instructions in the page Create sample PV and PVC (Link to https://oracle.github.io/weblogic-kubernetes-operator/samples/simple/storage/).
The users are recommended to change the values in the inputs file based on their requirement. The inputs file is located at "kubernetes/samples/scripts/create-weblogic-domain-pv-pvc/create-pv-pvc-inputs.yaml".
You need to ensure that the path mentioned for the property "weblogicDomainStoragePath" does exists (if not, you need to create it), has full access permissions set and also make sure that the folder does not contain any files.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 7]
```diff
@@ -0,0 +1,605 @@
...
+| `domainHome` | Home directory of the SOA domain. If not specified, the value is derived from the `domainUID` as `/shared/domains/<domainUID>`. | `/u01/oracle/user_projects/domains/soainfra` |
+| `domainPVMountPath` | Mount path of the domain persistent volume. | `/u01/oracle/user_projects/domains` |
+| `domainUID` | Unique ID that will be used to identify this particular domain. Used as the name of the generated WebLogic domain as well as the name of the Kubernetes domain resource. This ID must be unique across all domains in a Kubernetes cluster. This ID cannot contain any character that is not valid in a Kubernetes service name. | `soainfra` |
+| `domainType` | Type of the domain. Mandatory input for SOA Suite domains. You must provide one of the supported domain type values: `soa`,`osb`,`soaess`,`soaosb`, and `soaessosb`. | `soa`
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: For domainType filed, in the second column, please add the description of each domain type as below:

soa       : Deploys a SOA Domain
osb       : Deploys an OSB Domain (Oracle Service Bus)
soaess    : Deploys a SOA Domain with Enterprise Scheduler (ESS)
soaosb    : Deploys a Domain with SOA and OSB
soaessosb : Deploys a Domain with SOA, OSB and Enterprise Scheduler (ESS)

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 8]
```diff
@@ -0,0 +1,605 @@
...
+| `initialManagedServerReplicas` | Number of Managed Servers to initially start for the domain. | `2` |
+| `javaOptions` | Java options for starting the Administration Server and Managed Servers. A Java option can have references to one or more of the following pre-defined variables to obtain WebLogic domain information: `$(DOMAIN_NAME)`, `$(DOMAIN_HOME)`, `$(ADMIN_NAME)`, `$(ADMIN_PORT)`, and `$(SERVER_NAME)`. | `-Dweblogic.StdoutDebugEnabled=false` |
+| `logHome` | The in-pod location for the domain log, server logs, server out, and Node Manager log files. If not specified, the value is derived from the `domainUID` as `/shared/logs/<domainUID>`. | `/u01/oracle/user_projects/domains/logs/soainfra` |
+| `managedServerNameBase` | Base string used to generate Managed Server names. | `soa-server` |
+| `managedServerPort` | Port number for each Managed Server. | `8001` |
+| `namespace` | Kubernetes namespace in which to create the domain. | `soans` |
+| `persistentVolumeClaimName` | Name of the persistent volume claim. If not specified, the value is derived from the `domainUID` as `<domainUID>-weblogic-sample-pvc`. | `soainfra-domain-pvc` |
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: To be more specific for domain pv, please update:
Name of the persistent volume claim.
to
Name of the persistent volume claim created to host the domain home.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 9]
```diff
@@ -0,0 +1,605 @@
...
+lowercase letter and an underscore `("_")` is converted to a hyphen `("-")`.
+
+The sample demonstrates how to create a SOA Suite domain home and associated Kubernetes resources for a domain
+that has one cluster only. In addition, the sample provides the capability for users to supply their own scripts
+to create the domain home for other use cases. The generated domain YAML file could also be modified to cover more use cases.
+
+In addition, you should update the generated `domain.yaml` file by performing the following steps.
+
+##### Mandatory Configurations
+
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Please remove lines 237 to 243. These are obsolete now.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md [hunk 10]
```diff
@@ -0,0 +1,605 @@
...
+ In the generated `domain.yaml` file, update the `serverPod` definitions to set `precreateService` to `true`. You must apply this mandatory change.  See the sample `domain.yaml` file below.
+
+ ```
+ serverService:
+   precreateService: true
+ ```
+##### Optional Configurations
+
+Similarly, you can edit the `domain.yaml` file to add the resources section to the `serverPod` definition,
```
Comment (sbattagi) [inline docs-source/content/samples/simple/domains/soa-domain/_index.md]: Also please remove lines 246 to 264.

**Comment by marknelson** [comment]
> Thanks for checking @rjeberhard; @marknelson is right; there is still more work to do on it. I will remove WIP from the title when it is ready to merge.

Wrong Mark Nelson, be sure to use @markxnelson so I don't get weird notifications.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md (hunk 4 above)
Comment (rosemarymarano) [reply docs-source/content/samples/simple/domains/soa-domain/_index.md]: The name of that document is "Set up Kubernetes." However, I will change the link name to Kubernetes setup guide, as you wish.

File: docs-source/content/samples/simple/domains/soa-domain/_index.md (hunk 9 above)
Comment (rosemarymarano) [reply docs-source/content/samples/simple/domains/soa-domain/_index.md]: Wouldn't it make more sense to just remove this section if there are no mandatory configuration changes?

**Comment by rosemarymarano** [comment]
Samba,
I have incorporated your changes. Please review them.
Rosemary

**Pull request merged**